experimental/addons/dapr/definitions/dapr-publish-message-broker-trait-def.yaml

```diff
@@ -0,0 +1,37 @@
+apiVersion: core.oam.dev/v1beta1
+kind: TraitDefinition
+metadata:
+  name: dapr-publish-message-broker
+  namespace: vela-system
+spec:
+  appliesToWorkloads:
+  - deployments.apps
+  podDisruptive: true
+  schematic:
+    cue:
+      template: |
+        parameter: {
+          name: string
+          type: string
+          version: string
+          metadata: [string]: string
```

Comment:
metadata may not support case like the following:
```
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: pubsub
  namespace: default
spec:
  metadata:
  - name: redisHost
    value: redis-master:6379
  - name: redisPassword
    secretKeyRef:
      key: redis-password
      name: redis
  type: pubsub.redis
  version: v1
```
metadata item is not just string key-value pair , should be like this 
```
#MetadataItem: {
	name: string

	// +optional
	value?: #DynamicValue

	// +optional
	secretKeyRef?: #SecretKeyRef
}

// SecretKeyRef is a reference to a secret holding the value for the metadata item. Name is the secret name, and key is the field in the secret.
#SecretKeyRef: {
	name: string
	key:  string
}
```